compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -29,6 +29,64 @@ pub(super) fn opt_const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<
         let parent_node = tcx.hir().get(parent_node_id);
 
         match parent_node {
+            // This matches on types who's paths couldn't be resolved without typeck'ing e.g.
+            //
+            // trait Foo {
+            //   type Assoc<const N1: usize>;
+            //   fn foo() -> Self::Assoc<3>;
+            //   // note: if the def_id argument is the 3 then in this example
+            //   // parent_node would be the node for Self::Assoc<_>
+            // }
+            // We didnt write <Self as Foo>::Assoc so the Self::Assoc<_> is lowered to QPath::TypeRelative.
+            // I believe this match arm is only needed for GAT but I am not 100% sure - BoxyUwU
+            Node::Ty(hir_ty @ Ty { kind: TyKind::Path(QPath::TypeRelative(_, segment)), .. }) => {
+                // Walk up from the parent_node to find an item so that
+                // we can resolve the relative path to an actual associated type.
+                // For the code example above, this item would be the Foo trait.
```

Comment:
I found this comment confusing. I think what's going on is this:

* Find the item I that contains the anonymous constant so that we can create the context for it
* Using that context, we will convert the HIR for  `Self::Assoc<3>` into a type, which will be a fully resolved projection like `<Self as Foo>::Assoc<3>`

Reply:
Yea I wasn't great at this comment as I'm not all that familiar with the exact details of this Item stuff. What you said sounds correct to me so I'll just uh take that and put it in the comment x]